include/zephyr/dt-bindings/clock/stm32n6_clock.h

```diff
@@ -0,0 +1,193 @@
+#define STM32_SRC_IC3		(STM32_SRC_IC2 + 1)
+#define STM32_SRC_IC4		(STM32_SRC_IC3 + 1)
+#define STM32_SRC_IC5		(STM32_SRC_IC4 + 1)
+#define STM32_SRC_IC6		(STM32_SRC_IC5 + 1)
+#define STM32_SRC_IC7		(STM32_SRC_IC6 + 1)
+#define STM32_SRC_IC8		(STM32_SRC_IC7 + 1)
+#define STM32_SRC_IC9		(STM32_SRC_IC8 + 1)
+#define STM32_SRC_IC10		(STM32_SRC_IC9 + 1)
+#define STM32_SRC_IC11		(STM32_SRC_IC10 + 1)
+#define STM32_SRC_IC12		(STM32_SRC_IC11 + 1)
+#define STM32_SRC_IC13		(STM32_SRC_IC12 + 1)
+#define STM32_SRC_IC14		(STM32_SRC_IC13 + 1)
+#define STM32_SRC_IC15		(STM32_SRC_IC14 + 1)
+#define STM32_SRC_IC16		(STM32_SRC_IC15 + 1)
+#define STM32_SRC_IC17		(STM32_SRC_IC16 + 1)
+#define STM32_SRC_IC18		(STM32_SRC_IC17 + 1)
+#define STM32_SRC_IC19		(STM32_SRC_IC18 + 1)
+#define STM32_SRC_IC20		(STM32_SRC_IC19 + 1)
+
+/** Bus clocks */
+#define STM32_CLOCK_BUS_AHB1	0x250
+#define STM32_CLOCK_BUS_AHB2	0x254
+#define STM32_CLOCK_BUS_AHB3	0x258
+#define STM32_CLOCK_BUS_AHB4	0x25C
+#define STM32_CLOCK_BUS_AHB5	0x260
+#define STM32_CLOCK_BUS_APB1	0x264
+#define STM32_CLOCK_BUS_APB1_2	0x268
+#define STM32_CLOCK_BUS_APB2	0x26C
+#define STM32_CLOCK_BUS_APB3	0x270
+#define STM32_CLOCK_BUS_APB4	0x274
+#define STM32_CLOCK_BUS_APB4_2	0x278
+#define STM32_CLOCK_BUS_APB5	0x27C
+
+#define STM32_CLOCK_LP_BUS_SHIFT	0x40
+
+#define STM32_PERIPH_BUS_MIN	STM32_CLOCK_BUS_AHB1
+#define STM32_PERIPH_BUS_MAX	STM32_CLOCK_BUS_APB5
+
+#define STM32_CLOCK_REG_MASK	0xFFFU
+#define STM32_CLOCK_REG_SHIFT	0U
+#define STM32_CLOCK_SHIFT_MASK	0x1FU
+#define STM32_CLOCK_SHIFT_SHIFT	12U
+#define STM32_CLOCK_MASK_MASK	0x7U
+#define STM32_CLOCK_MASK_SHIFT	17U
+#define STM32_CLOCK_VAL_MASK	0x7U
+#define STM32_CLOCK_VAL_SHIFT	20U
+
+/**
+ * @brief STM32U5 clock configuration bit field.
+ *
+ * - reg   (1/2/3)         [ 0 : 11 ]
+ * - shift (0..31)         [ 12 : 16 ]
+ * - mask  (0x1, 0x3, 0x7) [ 17 : 19 ]
+ * - val   (0..7)          [ 20 : 22 ]
+ *
+ * @param reg RCC_CCIPRx register offset
+ * @param shift Position within RCC_CCIPRx.
+ * @param mask Mask for the RCC_CCIPRx field.
+ * @param val Clock value (0, 1, ... 7).
+ */
+#define STM32_DOMAIN_CLOCK(val, mask, shift, reg)					\
+	((((reg) & STM32_CLOCK_REG_MASK) << STM32_CLOCK_REG_SHIFT) |		\
+	 (((shift) & STM32_CLOCK_SHIFT_MASK) << STM32_CLOCK_SHIFT_SHIFT) |	\
+	 (((mask) & STM32_CLOCK_MASK_MASK) << STM32_CLOCK_MASK_SHIFT) |		\
+	 (((val) & STM32_CLOCK_VAL_MASK) << STM32_CLOCK_VAL_SHIFT))
+
+/** @brief RCC_CCIPRx register offset (RM0468.pdf) */
+#define CCIPR1_REG		0x144
+#define CCIPR2_REG		0x148
+#define CCIPR3_REG		0x14C
+#define CCIPR4_REG		0x150
+#define CCIPR5_REG		0x154
+#define CCIPR6_REG		0x158
+#define CCIPR7_REG		0x15C
+#define CCIPR8_REG		0x160
+#define CCIPR9_REG		0x164
+#define CCIPR12_REG		0x170
+#define CCIPR13_REG		0x174
+#define CCIPR14_REG		0x178
+
+/** @brief Device domain clocks selection helpers */
+/** CCIPR1 devices */
+#define ADF1_SEL(val)		STM32_DOMAIN_CLOCK(val, 7, 0, CCIPR1_REG)
```

Comment:
Add parentheses around `val` here and in  lines below.

Reply:
Done